components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/inbound/IdentityProcessor.java

```diff
@@ -139,9 +140,12 @@ public void init(InitConfig initConfig) {
         IdentityRequest identityRequest = context.getRequest();
 
         Map<String, String[]> parameterMap = identityRequest.getParameterMap();
-
-        parameterMap.put(FrameworkConstants.SESSION_DATA_KEY, new String[] { sessionDataKey });
-        parameterMap.put(FrameworkConstants.RequestParams.TYPE, new String[] { getName() });
+        Map<String, String[]> modParaMap = new HashMap<String, String[]>();
+        for (String key : parameterMap.keySet()) {
+            modParaMap.put(key, parameterMap.get(key));
+        }
+        modParaMap.put(FrameworkConstants.SESSION_DATA_KEY, new String[]{sessionDataKey});
+        modParaMap.put(FrameworkConstants.RequestParams.TYPE, new String[]{getName()});
 
         authenticationRequest.appendRequestQueryParams(parameterMap);
```

Comment:
Above changes are not required actually. The request parameters are passed to framework through AuthenticationRequest object in above line. It is put to cache and passed, so no need to pass as query parameters. Also passing as query parameters is vulnerable and that why we adopted the cache approach. The only issue that was there in my code is that I have unnecessarily tried to add FrameworkConstants.SESSION_DATA_KEY and FrameworkConstants.RequestParams.TYPE to an unmodifiable map which is not required at all.